cmake/strict/strict_flags_helpers.cmake

```diff
@@ -121,6 +124,10 @@ function(get_clang_base_flags flags)
     list(APPEND clang_flags "-Wno-packed")
     list(APPEND clang_flags "-Wno-padded")
     list(APPEND clang_flags "-Wno-reserved-id-macro")
+    # FIXME-WT-11788: Remove the following flag once the violation of the standard C11 7.1.3 has
+    # been addressed.
+    list(APPEND clang_flags "-Wno-reserved-identifier")
+    list(APPEND clang_flags "-Wno-unsafe-buffer-usage")
```

Comment:
Is that a flag we want to keep or is it a short term workaround as well?

Reply:
"Keep for the foreseeable future" is my current thinking - this flag is pretty broken right now (https://discourse.llvm.org/t/rfc-c-buffer-hardening/65734/85).

Reply:
The reserved identifier flag is fine as well, the identifiers in v7 are in no way less reserved. 😀